### PR TITLE
sdk: python: Fix provisioner

### LINKED ISSUE #3686: Fix engine provisioning
Depends on:
- https://github.com/dagger/dagger/issues/3653

## Todo

- [x] Use a worker thread with sync code[^1]
- [x] Follow instructions in https://github.com/dagger/dagger/issues/3653#issuecomment-1303984783
- [x] Use `docker.io/eriksipsma/test-dagger@sha256:eb6a15a52fed7ac51b26e181dc1c6b2731ba0451f1951c575d85690159cb0722`
- [ ] Test connection

[^1]: Quicker to reuse previous sync implementation and to have it ready for sync support.

### PR DESCRIPTION
**Pull request by helderco**
Fixes #3686

Credit to @sipsma for the implementation 🙌. 

Note: I converted to `sync` code to reuse in upcoming `sync` support. There's follow up work in #3687 to fix this.

Signed-off-by: Helder Correia

### CODE REVIEW

File: sdk/python/src/dagger/cli.py [hunk 1]
```diff
@@ -26,15 +27,13 @@ def generate(
...
-        if session.client.schema is None:
-            raise typer.BadParameter(
-                "Schema not initialized. Make sure the dagger engine is running."
-            )
-        code = codegen.generate(session.client.schema)
+    if cfg.host.scheme == "docker-image":
+        with Engine(cfg) as engine:
+            code = generate_code(engine.cfg)
+    else:
+        code = generate_code(engine.cfg)
```
Comment (sipsma) [inline sdk/python/src/dagger/cli.py:36]: @helderco I'm guessing this got messed up while I was rebasing, but I'm not sure what the original intention was 😔 Once this is taken care this should be good to merge.

Comment (helderco) [reply sdk/python/src/dagger/cli.py:36]: Yeah, the second one should be just `generate_code(cfg)` :)

**Comment by helderco** [comment]
@sipsma Thanks for getting that image ready 🙌

**Comment by helderco** [comment]
Ready for merge.

**Review by grouville — APPROVED** [review]
LGTM @helderco

**Pull request merged**